interpreter/llvm-project/llvm/include/llvm/ADT/PointerUnion.h

```diff
@@ -261,8 +261,9 @@ struct PointerLikeTypeTraits<PointerUnion<PTs...>> {
 
   // The number of bits available are the min of the pointer types minus the
   // bits needed for the discriminator.
-  static constexpr int NumLowBitsAvailable = PointerLikeTypeTraits<decltype(
-      PointerUnion<PTs...>::Val)>::NumLowBitsAvailable;
+  using nvhpcWorkAround = decltype(PointerUnion<PTs...>::Val);
```

Comment:
Is this patch a backport of the llvm upstream?

Reply:
No, LLVM upstream (17.0.6) does not have this change and nvc++ fails equally there. What is the procedure to apply such a workaround to ROOT's LLVM?

(while NVIDIA may be working on a fix to nvc++)

Reply:
I think it should be enough if we open a PR against LLVM mainline and backport it here. It'd be better to wait until it gets merged upstream but generally there is no need to.

cc: @hahnjo

Reply:
Either way, I'm not a fan of introducing a workaround to our copy of LLVM for a very specific compiler. I believe upstream has a very clear documentation on which compilers it supports, and NVIDIA's isn't one of them. It's not our job to fix this downstream.